lib/slack/real_time/concurrency/celluloid.rb

```diff
@@ -39,7 +39,7 @@ def run_loop
             logger.debug("#{self.class}##{__method__}") { e }
             driver.emit(:close, WebSocket::Driver::CloseEvent.new(1001, 'server closed connection')) unless @closing
           ensure
-            terminate
+            terminate if ::Celluloid::Actor.current.alive?
```

Comment:
I have tried `alive?`, `behavior_proxy.alive?`, `proxy.alive?`. All of them fail.
E.g:
```
NameError: undefined local variable or method `proxy' for #<Slack::RealTime::Concurrency::Celluloid::Socket:0x007fb7ee7d0d68>
```

Interestingly, when I inspect the object it behaves differently,

```ruby
+2.3.0 :016 > slack_client.instance_variable_get("@socket").inspect
 => "#<Celluloid::Proxy::Cell(Slack::RealTime::Concurrency::Celluloid::Socket:0x3fc194e19f0c)..
```

My guess is when there is an un-handled exception proxy bindings are removed. But it still doesn't explain why `terminate` method is still accessible.